### PR TITLE
Use generic Dockerfile for KEB jobs

### PR DESCRIPTION
**Pull request by wozniakjan**
**Description**

There is a newly introduced generic Dockerfile for KEB jobs - https://github.com/kyma-project/control-plane/pull/2475. This PR introduces the set of changes to use it so the old copy-paste Dockerfiles can be removed.

**Related issue(s)**

### CODE REVIEW

File: prow/jobs/control-plane/components/kyma-environment-broker/kyma-environment-broker-build.yaml [hunk 1]
```diff
@@ -65,7 +65,8 @@ presubmits: # runs on PRs
             args:
               - "--name=control-plane/kyma-environments-cleanup-job"
               - "--context=components/kyma-environment-broker"
-              - "--dockerfile=Dockerfile.cleanup"
+              - "--dockerfile=Dockerfile.job"
+              - "--build-arg=BIN=environmentscleanup"
```
Comment (halamix2) [inline prow/jobs/control-plane/components/kyma-environment-broker/kyma-environment-broker-build.yaml:69]: I don't think there is a `build-arg` flag.
 - The closest thing is still in PR: #6733

Comment (wozniakjan) [reply prow/jobs/control-plane/components/kyma-environment-broker/kyma-environment-broker-build.yaml:69]: thanks! I was under the impression it's raw kaniko building the images. I will put this as WIP and keep an eye on https://github.com/kyma-project/test-infra/pull/6733

Comment (Ressetkk) [reply prow/jobs/control-plane/components/kyma-environment-broker/kyma-environment-broker-build.yaml:69]: @wozniakjan not really. It's a wrapper for kaniko or BuildKit. image-builder doesn't support passing build args, however you can try to add them to the environment variables through the env-file. Look into https://github.com/kyma-project/test-infra/blob/88ddcc0f22eee0249af560c4647afb92aa15e7ba/development/image-builder/main.go#L376 and refer to the README.md file https://github.com/kyma-project/test-infra/blob/main/development/image-builder/README.md

Or you can use something else for building go-based containers.

Comment (wozniakjan) [reply prow/jobs/control-plane/components/kyma-environment-broker/kyma-environment-broker-build.yaml:69]: given https://github.com/kyma-project/test-infra/pull/6733 merged, @ralikio is going to take over this and change `--build-arg=` to -`-export-tags=`

**Comment by halamix2** [comment]
@dependabot rebase

**Review by ralikio — DISMISSED** [review]

**Review by Ressetkk — COMMENTED** [review]
Remember that your image will *always* be tagged with additional tag `image-name:$BIN`.

**Comment by ralikio** [comment]
After giving it a second thought I think you are right. Let's not build unnecessary images. Could please take a look at https://github.com/kyma-project/test-infra/pull/7023. Thank you @Ressetkk

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**